src/main/java/org/springframework/data/redis/connection/jedis/JedisConnectionFactory.java

```diff
@@ -866,8 +913,8 @@ private MutableJedisClientConfiguration getMutableConfiguration() {
 	}
 
 	private void assertInitialized() {
-		Assert.state(this.initialized, "JedisConnectionFactory was not initialized through afterPropertiesSet()");
-		Assert.state(!this.destroyed, "JedisConnectionFactory was destroyed and cannot be used anymore");
+		Assert.state(State.STARTED.equals(state.get()), "JedisConnectionFactory was not initialized through afterPropertiesSet()");
+		Assert.state(!State.STOPPED.equals(state.get()), "JedisConnectionFactory was destroyed and cannot be used anymore");
```

Comment:
We might want to reflect the lifecycle in the messages, as we want to allow starting after stopping, right?